docs/guides/basics/data-types.md

```diff
@@ -4,7 +4,287 @@ title: Data types
 
-[doc]: https://coda.io/d/Pack-Studio-Beta_dUBjm8jbi39/Building-with-the-SDK_suP0J#_luSAo
+[View Sample Code][samples]{ .md-button }
+
+
+## Value types
+
+The core, underlying type of a value is referred to as it's value type. The enumeration [`ValueType`][ValueType] defines the set of supported types, and they correspond to basic data types in JavaScript.
+
+For a formula the value type is specified in the `resultType` property, and for schemas it's specified in the `type` property. The type specified must match the type of the value returned by your code.
+
+```ts
+pack.addFormula({
+  // ...
+  resultType: coda.ValueType.String,
+  execute: async function ([], context) {
+    return "This is a string";
+  },
+});
+```
+
+[View all types][ValueType]{ .md-button }
+
+
+### Arrays
+
+Lists of data can be returned using the [`Array`][Array] value type. You must also specify the value type of the items in the array, using the `items` property.
+
+```ts
+pack.addFormula({
+  // ...
+  resultType: coda.ValueType.Array,
+  items: { type: coda.ValueType.String },
+  execute: async function ([], context) {
+    return ["This", "is", "a", "string", "array"];
+  },
+  // ...
+});
+```
+
+
+### Objects
+
+Structured data can be returned using the [`Object`][Object] value type. These objects must conform to an existing schema, as specified in the `schema` property. See the [Schemas guide][schemas] for more information about defining and using schemas.
+
+```ts
+const MySchema = coda.makeObjectSchema({
+  type: coda.ValueType.Object,
+  properties: {
+    property1: { type: coda.ValueType.String },
+    property2: { type: coda.ValueType.Number },
+    // ...
+  },
+  // ...
+});
+
+pack.addFormula({
+  // ...
+  resultType: coda.ValueType.Object,
+  schema: MySchema,
+  execute: async function ([], context) {
+    return {
+      property1: "This is a string",
+      property2: 42,
+      // ...
+    };
+  },
+});
+```
+
+
+## Value hints
+
+To indicate that Coda should display your value in a more meaningful way you can set a value hint. The enumeration [`ValueHintType`][ValueHintType] defines the set of supported value hints, which correspond to the [column formats][column_formats] in Coda. The value hint is set one using the `codaType` property of a formula or schema. Hints can only be used with certain value types, for example the `Percent` value hint can only be used with the `Number` value type.
+
+```ts
+pack.addFormula({
+  // ...
+  resultType: coda.ValueType.String,
+  codaType: coda.ValueHintType.Markdown,
+  execute: async function ([], context) {
+    return "This _is_ **markdown**";
+  },
+});
+```
+
+[View all types][ValueHintType]{ .md-button }
+
+### Markdown {. #markdown}
+
+The [`Markdown`][hint_markdown] value hint indicates that Coda should render the returned string as markdown. The value can contain [basic markdown syntax][markdown], but extensions used by other applications (like tables, emoji, etc) are not supported.
```

Comment:
Optional, just curious if "rendering as Markdown" (and similarly "rendering as html") is quite accurate. I see it as "interpret the value as Markdown and render it as rich text" but maybe this is too nuanced to be valuable to describe this way.

For example, HTML is not going to be rendered verbatim, it's going to be used as an input to a converter that outputs Coda's proprietary rich text format.

Reply:
Ya, that's a good callout. I think this updated text is more accurate while still being easy to understand.